src/bin/console/console.cpp

```diff
@@ -1078,6 +1086,58 @@ bool Console::_handle_rollback() {
   return false;
 }
 
+bool Console::_verify_plan(const std::vector<std::shared_ptr<AbstractLQPNode>>& lqp) {
```

Comment:
Why is this in the console? I'd expect all SQL interfaces to behave alike.

Moving this to the pipeline should also make it possible to do the check on a per-statement basis.

Reply:
And if the function were in the Pipeline, how would Hyrise behave then? What would the pipeine do differently than what it does now, which would be: Shut Hyrise down if the check failed? 

This is rightly placed in the console imo, because the console is the only place that we want to survive such input errors in, right?

Reply:
Use AssertInput and throw the exception that we are allowed to catch - same as syntactically incorrect SQL. The server should be able to handle failed AssertInputs just as well.

Reply:
Ah right, the server. Good, @d-justen, then we introduce `SQLPipelineStatement::_verify_lqp(...)` which performs AssertInput() checks - and call it from `get_unoptimized_plan()`

Reply:
Please keep the other open question (multi-statement queries) in mind. Those are ugly, but we shouldn’t break them.

Reply:
@mrks By moving the method to `SQLPipelineStatement` it doesn't break multi-statement queries since every statement is verified one after another